EarlGrey/Action/GREYMultiFingerSwipeAction.m

```diff
@@ -0,0 +1,173 @@
+}
+    
+- (instancetype)initWithDirection:(GREYDirection)direction
+                         duration:(CFTimeInterval)duration
+                  numberOfFingers:(NSUInteger)numberOfFingers {
+    
+    return [self initWithDirection:direction
+                          duration:duration
+                    numberOfFingers:numberOfFingers
+                      percentPoint:CGPointMake(0.5, 0.5)];
+}
+    
+    
+- (instancetype)initWithDirection:(GREYDirection)direction
+                         duration:(CFTimeInterval)duration
+                  numberOfFingers:(NSUInteger)numberOfFingers
+                    startPercents:(CGPoint)startPercents {
+    
+    return [self initWithDirection:direction
+                          duration:duration
+                   numberOfFingers:numberOfFingers
+                      percentPoint:startPercents];
+}
+    
+#pragma mark - GREYAction
+    
+- (BOOL)perform:(id)element error:(__strong NSError**)errorOrNil {
+    if (![self satisfiesConstraintsForElement:element error:errorOrNil]) {
+        return NO;
+    }
+    
+    UIWindow *window = [element window];
+    if (!window) {
+        if ([element isKindOfClass:[UIWindow class]]) {
+            window = (UIWindow *)element;
+        } else {
+            NSString *errorDescription =
+            [NSString stringWithFormat:@"Cannot multi finger swipe on view [V], as it has no window and "
+             @"it isn't a window itself."];
+            NSDictionary *glossary = @{ @"V" : [element grey_description]};
+            GREYError *error;
+            error = GREYErrorMake(kGREYSyntheticEventInjectionErrorDomain,
+                                  kGREYOrientationChangeFailedErrorCode,
+                                  errorDescription);
+            error.descriptionGlossary = glossary;
+            if (errorOrNil) {
+                *errorOrNil = error;
+            } else {
+                [GREYAssertions grey_raiseExceptionNamed:kGREYGenericFailureException
+                                        exceptionDetails:@""
+                                               withError:error];
+            }
+            
+            return NO;
+        }
+    }
+    
+    NSMutableArray *multiTouchPaths = [[NSMutableArray alloc] init];
+    CGRect accessibilityFrame = [element accessibilityFrame];
+    
+    for(NSUInteger i = 0; i < _numberOfFingers; i++) {
+        
+        CGFloat xOffset, yOffset;
+        switch (_direction) {
+            case kGREYDirectionDown:
+            case kGREYDirectionUp:
+                xOffset = (CGFloat)i * 10;
+                yOffset = 0.0;
+                break;
+            
+            case kGREYDirectionLeft:
+            case kGREYDirectionRight:
+                xOffset = 0.0;
+                yOffset = (CGFloat)i * 10;
+                break;
+        }
+        
+        CGPoint startPoint =
+        CGPointMake(accessibilityFrame.origin.x + accessibilityFrame.size.width * _startPercents.x + xOffset,
```

Comment:
oh my bad...
I meant introducing some brackets to group the operands like:
`CGPointMake(a + (b * c) + d, p + (q / r))`